src/components/Form.js

```diff
@@ -45,12 +45,10 @@ export default class Form extends Component {
         errors: state.errors || [],
         errorSchema: state.errorSchema || {}
       };
-    const idSchema = toIdSchema(schema, uiSchema["ui:rootFieldId"], definitions);
```

Comment:
Where is this `getStateFromProps` called, and does that call expect an `idSchema` to be in this object? Does it make sense to do the same kind of thing you did in `render` and return either the `props.idSchema` or this `toIdSchema()` result?

Reply:
`getStateFromProps` is only called in the constructor and in the `componentWilReceiveProps` lifecycle method:

![image](https://user-images.githubusercontent.com/2008881/92286249-54f72700-eebb-11ea-9f2d-2a4ae603f408.png)

The only place `idSchema` is used is puling it out of the state to be used in the render method, so I've just shifted things around and decided to get it from `props` instead of copying it into state, and _then_ getting it back out again.

The only possible reason that having `idSchema` in the `state` could matter is in the calls to `onChange` and `onSubmit`, since theoretically there could be a callback that used that object.  I'll add it back in I guess :upside_down_face: